x/staking/keeper/querier.go

```diff
@@ -317,7 +317,17 @@ func queryRedelegations(ctx sdk.Context, req abci.RequestQuery, k Keeper) ([]byt
 }
 
 func queryPool(ctx sdk.Context, k Keeper) ([]byte, sdk.Error) {
-	pool := k.GetPool(ctx)
+	bondDenom := k.BondDenom(ctx)
+	bondedPool := k.GetBondedPool(ctx)
+	notBondedPool := k.GetNotBondedPool(ctx)
+	if bondedPool == nil || notBondedPool == nil {
+		return nil, sdk.ErrInternal("pool accounts haven't been set")
+	}
+
+	pool := types.NewPool(
```

Comment:
I think we should eliminate the pool type altogether not that we have module accounts - it's used in no place besides these queries - and module account queries should all go through the supply keeper directly anyways........ do we have supply keeper queries yet btw?

Reply:
But the supply querier won’t be able to query the bonded and not bonded supply tho. I left the type and the pool querier bc it fulfills that purpose (and to not break additional api endpoints)

Reply:
yeah like we should deprecate it - maybe we can add in some comments to the cli that this will be deprecated

Reply:
Oh yeah I forgot this is the whole unbonded supply - fine to keep it